library/src/config.rs

```diff
@@ -60,14 +62,14 @@ pub fn testing_reset_config() {
     });
 }
 
-pub fn check_initialized_and_call<F, R>(f: F, config: &ResolvedConfig) -> R
+pub fn check_initialized_and_call<F, R>(f: F, config: &ResolvedConfig) -> anyhow::Result<R>
 where
     F: FnOnce(&ResolvedConfig) -> R,
```

Comment:
To avoid the Result<Result<T>> we can specify that F returns a Result no?

Reply:
If I understand correctly, updating the constraints on `F` to return `Result<R>` instead of `R` will take us in the wrong direction (it will add an extra layer of `Result`). 

`check_initialized_and_call` needs to return a `Result`, possibly without invoking `F`, because it bails if `config` is not initialized. `R` _can_ be a `Result` (and sometimes is), but the code is usually cleaner if it isn't (see `check_for_update`, which would need to return `Ok(false)` instead of just `false` if an update is not available).

Reply:
Updated as per offline convo